File: AlgorithmTest/Programmers/src/lv2/스킬트리.java
package lv2;
import java.util.*;
public class 스킬트리 {

    public static void main(String[] args) {
        스킬트리 app = new 스킬트리();
        String sk="CBD";
        String [] sTree = {"BACDE","CBADF","AECB","BDA"};
        System.out.println(app.solution(sk, sTree));
    }

    public int solution(String skill, String[] skill_trees) {
        int answer = 0;
        int [] chk = new int[skill.length()];
        for(String temp : skill_trees) {
            int flag = 1;
            for(int i=0;i<skill.length();i++) {
                chk[i]=temp.indexOf(skill.charAt(i));
                System.out.print(chk[i]+" ");
            }
            for(int i=0;i<skill.length()-1;i++) {
                if((chk[i]==-1&&chk[i+1]!=-1) || chk[i+1]!=-1 && chk[i]>chk[i+1] ) {
                    flag = 0;
                    break;
                }

            }
            System.out.println();
            if(flag==1)answer++;
        }
        return answer;
    }

}

File: Python/Programmers/Lv2/스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    for s in skill_trees:
        temp = []
        flag = True
        for i in skill:
            temp.append(s.find(i))
        for i in range(len(temp)-1):
            if (temp[i] == -1 and temp [i+1] > -1 ) or temp[i]>temp[i+1]>=0:
                flag = False
                break
        if (flag):
            print(s)
            answer+=1
            
    return answer

skill = 'CBD'
skill_t = ["BACDE", "CBADF", "AECB", "BDA"]

n = solution(skill,skill_t)
print(n)

def solution2 (skill,skill_trees):
    answer = 0
    for skills in skill_trees:
        skill_list = list(skill)
        for s in skills:
            if s in skill:
                if s != skill_list.pop(0):
                    break
        else:
            answer+=1

    return answer

# for - else 문은 for문이  끝까지 간다면 else문을 실행시킨다.